core/domain/services/admin/events/editor/EventEditor.php

```diff
@@ -0,0 +1,159 @@
+use EventEspresso\core\domain\services\assets\EspressoEditorAssetManager;
+use EventEspresso\core\exceptions\InvalidDataTypeException;
+use EventEspresso\core\exceptions\InvalidInterfaceException;
+use EventEspresso\core\exceptions\ModelConfigurationException;
+use EventEspresso\core\exceptions\UnexpectedEntityException;
+use EventEspresso\core\services\assets\JedLocaleData;
+use InvalidArgumentException;
+use ReflectionException;
+use WP_Post;
+use WPGraphQL\Router;
+
+/**
+ * Class EventEditor
+ * Description
+ *
+ * @package EventEspresso\core\domain\services\admin\events\editor
+ * @author  Manzoor Wani,  Brent Christensen
+ * @since   $VID:$
+ */
+class EventEditor
+{
+
+    /**
+     * @var EE_Admin_Config
+     */
+    protected $admin_config;
+
+    /**
+     * @var CurrentUser $current_user
+     */
+    protected $current_user;
+
+    /**
+     * @var EE_Event
+     */
+    protected $event;
+
+    /**
+     * @var EventEditorGraphQLData
+     */
+    protected $event_editor_gql_data;
+
+    /**
+     * @var GeneralSettings $general_settings
+     */
+    protected $general_settings;
+
+    /**
+     * @var JedLocaleData $jed_locale
+     */
+    private $jed_locale;
+
+
+    /**
+     * EventEditor constructor.
+     *
+     * @param EE_Admin_Config $admin_config
+     * @param CurrentUser $current_user
+     * @param EE_Event        $event
+     * @param EventEditorGraphQLData $event_editor_gql_data
+     * @param GeneralSettings $general_settings
+     * @param JedLocaleData   $jed_locale
+     */
+    public function __construct(
+        EE_Admin_Config $admin_config,
+        CurrentUser $current_user,
+        EE_Event $event,
+        EventEditorGraphQLData $event_editor_gql_data,
+        GeneralSettings $general_settings,
+        JedLocaleData $jed_locale
+    ) {
+        $this->admin_config = $admin_config;
+        $this->current_user = $current_user;
+        $this->event = $event;
+        $this->event_editor_gql_data = $event_editor_gql_data;
+        $this->general_settings = $general_settings;
+        $this->jed_locale = $jed_locale;
+        add_action('admin_enqueue_scripts', [$this, 'loadScriptsStyles']);
+    }
+
+
+    /**
+     * @throws EE_Error
+     * @throws InvalidArgumentException
+     * @throws InvalidDataTypeException
+     * @throws InvalidInterfaceException
+     * @throws ModelConfigurationException
+     * @throws ReflectionException
+     * @throws UnexpectedEntityException
+     * @throws DomainException
+     * @since $VID:$
+     */
+    public function loadScriptsStyles()
+    {
+        if ($this->admin_config->useAdvancedEditor()) {
+            $eventId = $this->event instanceof EE_Event ? $this->event->ID() : 0;
+            if (! $eventId) {
+                global $post;
+                $eventId = isset($_REQUEST['post']) ? absint($_REQUEST['post']) : 0;
+                $eventId = $eventId === 0 && $post instanceof WP_Post && $post->post_type === 'espresso_events'
```

Comment:
this `0` make me think that another intermediary variable might be helpful here because it doesn't make a lot of sense to set `$eventId` to zero and the check `$eventId === 0`
also, there is a big chance that I might not see the big picture here

Reply:
will do